### ISSUE TITLE
Frame Pacing Issue

### ISSUE THREAD

**Issue by Peduls** [citra-emu/citra#6193]
So I noticed a couple of my 3DS games seemed to not be running quite as smooth as they used to despite showing 100% 60FPS so I tried downgrading a few versions to see if it was just my imagination or not.
I found that nightly-1790 is the first build that exhibits this issue. The status bar always shows the game running at 100% 60FPS but from  build nightly-1790 and on there seems to be some kind of frame pacing issue that makes the game look like it's stuttering constantly.
The games I have noticed it in so far are Animal Crossing: New Leaf and BRAVELY DEFAULT.

**Comment by Peduls** [comment]
I just tested to see if I would experience the same micro-stutter/pacing issue with the Vulkan build and the issue is gone even when using OpenGL with that build.

**Comment by Peduls** [comment]
I don't think it actually first appeared in Nightly-1790, but rather the stuttering became far more noticeable after this build.

**Comment by Peduls** [comment]
Disabling VSync through Citra and forcing Vertical sync and Triple buffering through NVIDIA Control Panel also reduces the stuttering effect significantly.

**Comment by BParks21** [comment]
The frame Pacing is absolutely terrible in the nightly and canary atm. Every game stutters. Even though the emulator reports a locked 60FPS the frame times are in the 40's, 30's and 20's. These frame times are more in line with a 30 fps game. If the game is outputting 60fps it should be much closer to 16.6ms frame times. It's simply broken. Unplayable imo. Please fix this!

**Comment by SentaiBrad** [comment]
I have noticed a frame pacing inconsistency as well. I can see this in action with Fantasy Life on my PC and Steam Deck, and the Steam Deck even shows a frame pacing graph that illustrates this further. This used not to be the case. 

Is any log or info I can provide? On Canary 2387, but I've noticed this behavior for quite a while.

**Comment by nickxcom** [comment]
Can confirm consistent frame pacing problems in all versions post-"nightly 1789", as far as I can tell. That's the last one I've found that does not exhibit these frame pacing problems/stutter at full speed. Nightly 1789 appears to perform perfectly.

**Comment by SirDevil99** [comment]
Pretty bad frame pacing for sure.
Will this be fixed or nah?

**Comment by kyle-kc** [comment]
Can confirm that I'm seeing this issue as well. Reverting to 1789 fixes the stutter but of course it's lacking many of the recent features.

**Comment by kyle-kc** [comment]
For whatever reason, building and running from source doesn't seem to have this issue. I only see the issue when using the version installed by the Citra installer.

**Comment by BParks21** [comment]
Thats weird, what kind if frame times do you see when using it from building it? I've heard that the vulkan experimental builds or whatever don't have the issue.

**Comment by kyle-kc** [comment]
Sorry I'm not sure, but I can see that the stuttering is gone completely when I build from source.

**Comment by Peduls** [comment]
Okay, I might have discovered something helpful. I just tested the experimental Vulkan branch again and I noticed the stuttering was present. I didn't know how this was possible since I know it didn't stutter that last time I tested the Vulkan build. Then I realized I had downloaded the windows-mingw artifact instead of the msvc artifact. The windows-mingw build has stuttering and the Citra installer makes a "nightly-mingw" in AppData/Local/Citra not "nightly-msvc" I don't know anything about building from source so forgive me if I sound stupid, but I just thought this might be helpful info.

**Comment by FearlessTobi** [comment]
That actually makes sense, as the changes between Nightly 1789 and 1790 were to our build system.
https://github.com/citra-emu/citra/pull/6148 @SachinVin Any idea how this could have caused the issues?

**Comment by Peduls** [comment]
The latest nightly release (nightly-1855) doesn't include a msvc asset so I can no longer circumvent this issue.

**Comment by Immersion95** [comment]
Same thing happening here on a 120hz screen with Freesync

**Comment by BParks21** [comment]
It does look smoother now actually. However msi after burner is still showing strange numbers for the frame times. The frame times are always in the 20's for some reason. Before they were always in the 20's but it didn't look smooth at all. Now they are still in the 20's but what I'm seeing is way smoother. Something is still off, there are still minor little stutters I can see here and there.

**Comment by v1993** [comment]
Running Canary 2575's AppImage on linux nvidia; game Kirby Planet Robobot:

* OpenGL shows steady 60FPS framerate and mangohud indicates consistent frametimes, but the actual result stutters like hell
* Vulkan runs much better, although there are still minor stutters here and there

**Comment by BParks21** [comment]
> Running Canary 2575's AppImage on linux nvidia; game Kirby Planet Robobot:
> 
> * OpenGL shows steady 60FPS framerate and mangohud indicates consistent frametimes, but the actual result stutters like hell
> * Vulkan runs much better, although there are still minor stutters here and there

This is my exact observation.
